docs/advanced.md

```diff
@@ -473,6 +473,28 @@ var argv = require('yargs')
   .argv;
 ```
 
+#### Global Pre-check Middleware
+
+Middleware can be added (and processed) prior to any validation checks but after parsing so required fields maybe implicitly obtained from other methods (such as searching config files, looking for specific files in the current working directory), etc. These can only be set globally at the moment.
```

Comment:
maybe -> may be

Reply:
I rebased this and resolved the merge conflicts, tests are passing, I also changed the text as requested.  

To @bcoe's point the pre-checks-middleware does not support asynchronous functions, where as the middleware can. This is because the pre-checks-middleware must change the innerArgv object in the flow of the command run, and before the middleware is ran. If there's ever an intent to get the pre-check middleware to support async it would require drastic changes to the flow of runCommand and my bet is the rest of yargs.  

Maybe it would be useful to rename pre-checks-middleware to something else to prevent confusion, and for now, disallow asynchronous functions for pre-checks-middleware?